src/ledger/AccountFrame.cpp

```diff
@@ -104,9 +104,12 @@ bool
 AccountFrame::isValid()
 {
     auto const& a = mAccountEntry;
-    return isString32Valid(a.homeDomain) && a.balance >= 0 &&
-           std::is_sorted(a.signers.begin(), a.signers.end(),
-                          &AccountFrame::signerCompare);
+    bool res = isString32Valid(a.homeDomain);
+    res = res && a.balance >= 0;
+    res = res && std::is_sorted(a.signers.begin(), a.signers.end(),
+                                &AccountFrame::signerCompare);
+    res = res && ((a.flags & ~MASK_ACCOUNT_FLAGS) == 0);
+    return res;
```

Comment:
This does not implement all of the checks that are applied in makeValid, including:
- signers must be unique
- weights of signers cannot be 0

Reply:
* is_sorted does ensure that signers are unique (strict comparison)
* weight - yep, I am adding this check

Reply:
yikes, you are correct, every word is important `non-descending` means that it's using `!(a < b)` internally